### PR TITLE
Implement an infallible version of `Payload::read()` to reduce unwraps.

### PR DESCRIPTION
**Pull request by briansmith**

### CODE REVIEW

File: rustls/src/msgs/base.rs [hunk 1]
```diff
@@ -23,6 +23,10 @@ impl Payload {
     pub fn empty() -> Payload {
         Payload::new(Vec::new())
     }
+
+    pub fn read(r: &mut Reader) -> Self {
```
Comment (djc) [inline rustls/src/msgs/base.rs:27]: Maybe a different name would make the distinction between this and `Codec::read()` more obvious?

**Review by djc — COMMENTED** [review]
Nice!

File: rustls/src/msgs/base.rs (hunk 1 above)
Comment (briansmith) [reply rustls/src/msgs/base.rs:27]: I think it's fine this way, but if you suggest a better name I'll change it. read_infallible?

Comment (djc) [reply rustls/src/msgs/base.rs:27]: Maybe `Codec::read()` should become `try_read()` (though that should go in a different PR). What do you think?

**Pull request merged**